root/static/scripts/edit/components/edit/FullChangeDiff.js

```diff
@@ -8,15 +8,24 @@
  */
 
 import * as React from 'react';
+import _ from 'lodash';
 
-import {type DiffProps} from './Diff';
+type FullChangeDiffProps = {
+  +label: string,
+  +newContent: Object,
+  +oldContent: Object,
+};
 
-const FullChangeDiff = ({label, newText, oldText}: DiffProps) => (
-  oldText === newText ? null : (
+const FullChangeDiff = ({
+  label,
+  newContent,
+  oldContent,
+}: FullChangeDiffProps) => (
+  _.isEqual(oldContent, newContent) ? null : (
```

Comment:
Deep compare is don't because the input to FullChangeDiff can also be a text in some cases

Reply:
I'm not sure I follow. You shouldn't ever need to deeply compare objects for this - can you give a concrete example?

Reply:
Case 1 :

> <FullChangeDiff
>           label={addColonText(l('Type'))}
>           newContent={display.type.new ? display.type.new.name : ''}
>           oldContent={display.type.old ? display.type.old.name : ''}
>         />

Case 2 :

> <FullChangeDiff
>             label={addColonText(l('Area'))}
>             newContent={area.new
>               ? <DescriptiveLink entity={area.new} />
>               : ''}
>             oldContent={area.old
>               ? <DescriptiveLink entity={area.old} />
>               : ''}
>           />

Case 1 is from EditAlias where the content type is String, whereas in Case 2 which is from EditPlace the content type is Object where gid had to be compared. I'll currently use conditions using type method and push the changes, please let me know if you have a better idea

Reply:
Just display `null` if `area.new.gid === area.old.gid`. :) To clarify I wasn't sure what you meant by "If I'd have used .gid to compare then the object structure with gid would have to be enforced." We don't care about the rest of the object here. If the gids are equal, the rest of the area properties will be equal too - you can't change the properties of an area in a place edit. If there is a case where we ever need to deeply compare objects, I'd rather that be done outside the component and not by default, because it's a relatively expensive operation.

Reply:
Yep made the changes and fixed the test